### PR TITLE
Exclude copybara branch from the Generate workflow

### PR DESCRIPTION
**Pull request by marcgovi**
The `generate` workflow runs for any `push` or any `pull request` action and makes sure no changes have been introduced to the proto schemas.
The problem is that `copybara` branch might in fact be performing changes to those schemas (eg: adding a new one), thus making it fail, and not allowing admins to merge it without disabling branch protection first.

I'm proposing to skip this workflow for the `copybara` branch to reduce manual intervention.
FYI, once a `copybara` PR is submitted to `main`, the generator will run anyway and create a PR with any new changes.

### CODE REVIEW

**Comment by grayside** [comment]
My understanding is this diff check is in place primarily to make sure humans are paying attention to potential compatibility breaks. Is a possibly better path to only error if there are deletions or changes but not on pure additions?

@grant do you have insight on the role of the diff you could share?

**Review by grayside — CHANGES_REQUESTED** [review]
Blocking merge to make sure (minimally) I understand the risks of this change.

**Review by grant — APPROVED** [review]
Great, this is an improvement. See some background below:

Happy to add perspective.

Re Adam:

> My understanding is this diff check is in place primarily to make sure humans are paying attention to potential compatibility breaks. Is a possibly better path to only error if there are deletions or changes but not on pure additions?
> 
> @grant do you have insight on the role of the diff you could share?

This GitHub workflow was mainly to add a ✅ check to ensure that any generated resources were created along with the pull request change (for example, changing the JSON schema generator / catalog generator / README generator).

Without this check, there were additional PRs to sync `main`, making me send lots of PR.

The `copybara` branch is special compared to all other branches, where it is not expected to have generated resources, so this PR lgtm.

---

For testing proto incompatibilities, we don't have an exact test that checks API-level incompatibilities.

The best related thing we have for that is testing the sample JSON payloads which test against our protos:

https://github.com/googleapis/google-cloudevents/tree/main/testdata

Maybe there's a proto diff we could add - that should probably be a g3 presubmit error though and not downstream here (we can't do much at that point).

Note: Make sure to check with Adam before going with this PR.

**Comment by grayside** [comment]
> Maybe there's a proto diff we could add - that should probably be a g3 presubmit error though and not downstream here (we can't do much at that point).

Definitely, I see it as a defense in-depth discussion, but also making sure I understand the pieces in play.

**Review by grayside — APPROVED** [review]

**Pull request merged**
